### PR TITLE
UI Improvements

### PR DESCRIPTION
**Pull request by matthieu-fesselier**
- Clean design files structure
- Improve Bootstrap usage
- Various design fixes
- Typography
- Improve responsive version
- RTL version

### CODE REVIEW

File: assets/js/matching.js [hunk 1]
```diff
@@ -82,17 +82,17 @@ $(document).ready(() => {
         const menteeTable = $('#selected_international');
 
         if (dataTable.length) {
+            dataTable.on('draw.dt', () => {
```
Comment (victor-paumier) [inline assets/js/matching.js]: listener before init?

File: assets/sass/_vendors.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+@import "~bootstrap-v4-rtl/scss/close";
+@import "~bootstrap-v4-rtl/scss/modal";
+@import "~bootstrap-v4-rtl/scss/tooltip";
+@import "~bootstrap-v4-rtl/scss/utilities";
+
+@import "~bootstrap-v4-rtl/scss/rtl";
+
+
+
+/***
```
Comment (JeremieSamson) [inline assets/sass/_vendors.scss]: Why this all comment block ?
If it's not needed it probably should not be committed

File: assets/sass/components/_dashboard.scss [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      p{
+        margin-bottom: 0;
+        margin-top: 0.25rem;
+      }
+    }
+  }
+
+  &__form{
+    @include media-breakpoint-down('sm'){
+      padding-left: 1rem !important;
```
Comment (JeremieSamson) [inline assets/sass/components/_dashboard.scss]: why do you need to add `!important` here ? :thinking:

File: assets/sass/components/_dashboard.scss [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    top: 0;
+    font-weight: $bold;
+    width: 110px;
+    line-height: 16px;
+    font-size: 13px;
+
+    @include media-breakpoint-down('sm'){
+      width: 100%;
+      top: -20px;
+      text-align:center !important;
```
Comment (JeremieSamson) [inline assets/sass/components/_dashboard.scss]: same question about `!important`

File: assets/sass/components/_datatable.scss [hunk 5]
```diff
@@ -1,4 +1,76 @@
...
+  }
+}
+
+// Pagination
+.dataTables_wrapper .dataTables_paginate .paginate_button{
+  background: transparent;
+  padding: 0.3em 0.8em;
+
+  &:hover{
+    background: #333;
```
Comment (victor-paumier) [inline assets/sass/components/_datatable.scss]: color variable?

File: assets/sass/components/_form.scss [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+// Required fields
+label.required:after {
+  content: " *";
+}
+
+// Forms white box
+.white-box {
+  border-radius: 10px;
+  background-color: white;
```
Comment (JeremieSamson) [inline assets/sass/components/_form.scss]: can you put the color from a variable please ?even if it's called `$white`

File: assets/sass/components/_homepages.scss [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+    font-size: 19px;
+    line-height: 23px;
+    margin-top: 15px;
+  }
+  &__row {
+    height: 100%;
+  }
+  &__background{
+    position: relative;
+    background: rgba(0,0,0,0.2);
```
Comment (JeremieSamson) [inline assets/sass/components/_homepages.scss]: What do you think about putting a `$var` instead of rgba and a second property `opacity` here ?
Would be more readable IMO

File: package.json [hunk 8]
```diff
@@ -33,6 +33,7 @@
     "bazinga-translator": "^2.6.6",
     "bootstrap": "^4.3.1",
```
Comment (victor-paumier) [inline package.json]: is this one still in use?

File: templates/Dashboard/Stats/bc_view.html.twig [hunk 9]
```diff
@@ -34,17 +34,9 @@
...
-</div>
-
-{% block stylesheets %}
-    <link rel="stylesheet" href="{{ asset('build/dashboard_insight.css') }}" type="text/css">
-{% endblock %}
-
-{% block javascripts %}
-    <script src="{{ asset('build/dashboard_insight.js') }}"></script>
-{% endblock %}
+</div>
```
Comment (JeremieSamson) [inline templates/Dashboard/Stats/bc_view.html.twig]: Line break here plz

File: templates/Dashboard/Stats/bc_view.html.twig [hunk 10]
```diff
@@ -34,17 +34,9 @@
             </ul>
         </div>
     </div>
-    <hr class="lightblue">
+    <hr>
 
     <div id="stats-content">
         {% include 'Dashboard/Stats/bc_view_content.html.twig' %}
     </div>
-</div>
```
Comment (victor-paumier) [inline templates/Dashboard/Stats/bc_view.html.twig]: Why did you move this import to the Dashboard/index?

File: templates/Hobby/index.html.twig [hunk 11]
```diff
@@ -16,50 +14,51 @@
...
-            </thead>
-            <tbody>
-            {% for hobby in hobbies %}
-                <tr data-id="{{ hobby.id }}">
-                    <td>{{ hobby.name }}</td>
-
+        <div class="mt-3">
+            <table class="table table-striped table-hover nowrap custom-datatable" id="hobbies">
+                <thead>
+                <tr class="bg-white">
```
Comment (JeremieSamson) [inline templates/Hobby/index.html.twig]: indentat please

File: templates/Motivation/index.html.twig [hunk 12]
```diff
@@ -16,52 +14,54 @@
...
-            </thead>
-            <tbody>
-            {% for motivation in motivations %}
-                <tr data-id="{{ motivation.id }}">
-                    <td>{{ motivation.name }}</td>
-
+        <div class="mt-3">
+            <table class="table table-striped table-hover nowrap custom-datatable" id="motivations">
+                <thead>
+                <tr class="bg-white">
```
Comment (JeremieSamson) [inline templates/Motivation/index.html.twig]: indentation plz

File: assets/js/matching.js (hunk 1 above)
Comment (matthieu-fesselier) [reply assets/js/matching.js]: Yes otherwise the `draw.dt` event is sent after init but the event listener is not defined yet.

File: assets/sass/_vendors.scss (hunk 2 above)
Comment (matthieu-fesselier) [reply assets/sass/_vendors.scss]: This block contains all the bootstrap modules not imported yet. If the developer wants to use one, he just has to take the line from here, which I find more convenient than going in `node_modules/...` to check the available modules.
If you prefer, I can still remove this.

File: assets/sass/components/_dashboard.scss (hunk 3 above)
Comment (matthieu-fesselier) [reply assets/sass/components/_dashboard.scss]: Because we use the bootstrap `px-5` class in the HTML, which use `!important` to add padding. To override for small screens, we need this.

File: assets/sass/components/_dashboard.scss (hunk 4 above)
Comment (matthieu-fesselier) [reply assets/sass/components/_dashboard.scss]: Same here, we use the bootstrap classes `text-right` and `text-left` in the HTML. To override, we need `!important`

File: assets/sass/components/_homepages.scss (hunk 7 above)
Comment (matthieu-fesselier) [reply assets/sass/components/_homepages.scss]: It won't work here because there is some content in this element, which will be affected by the `opacity`.
However, to be more readable, I can do: `background: rgba($black,0.2);`

File: templates/Dashboard/Stats/bc_view.html.twig (hunk 10 above)
Comment (matthieu-fesselier) [reply templates/Dashboard/Stats/bc_view.html.twig]: Before we had all the styles in the CSS file `dashboard_insight` (including bootstrap).  
I changed it to make profit of the browser cache, to have a `main.css` which stays the same on every pages, and some small specific CSS files on different pages.
To do that, I needed to call `{{ parent() }}` in the `stylesheets` block to import the `main`, but it can only be done in a child template, so I had to move it in `index`.  

But it's true that this code is not used when the user is not a coordinator, so maybe we can add one test to import it only in this case?

File: assets/sass/components/_datatable.scss (hunk 5 above)
Comment (matthieu-fesselier) [reply assets/sass/components/_datatable.scss]: Done in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: package.json (hunk 8 above)
Comment (matthieu-fesselier) [reply package.json]: Not used indeed, removed in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: assets/sass/components/_form.scss (hunk 6 above)
Comment (matthieu-fesselier) [reply assets/sass/components/_form.scss]: Done in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: templates/Dashboard/Stats/bc_view.html.twig (hunk 9 above)
Comment (matthieu-fesselier) [reply templates/Dashboard/Stats/bc_view.html.twig]: Done in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: templates/Hobby/index.html.twig (hunk 11 above)
Comment (matthieu-fesselier) [reply templates/Hobby/index.html.twig]: Done in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: templates/Motivation/index.html.twig (hunk 12 above)
Comment (matthieu-fesselier) [reply templates/Motivation/index.html.twig]: Done in 0fbaf0fa286dd95d7d07f80440d66c72edabbe00

File: assets/js/matching.js (hunk 1 above)
Comment (victor-paumier) [reply assets/js/matching.js]: Ok, but why is it not the same order in other datatable init? Is that because this particular datatable is not filled using ajax?

Comment (matthieu-fesselier) [reply assets/js/matching.js]: I just changed this one because the click event was not working anymore, I didn't look at the others. But maybe it should be safer to do the same on the others.

File: assets/sass/_vendors.scss (hunk 2 above)
Comment (JeremieSamson) [reply assets/sass/_vendors.scss]: What do you mean by `not imported yet` :thinking: ?

File: assets/sass/components/_dashboard.scss (hunk 3 above)
Comment (JeremieSamson) [reply assets/sass/components/_dashboard.scss]: Ok :)

File: assets/sass/components/_homepages.scss (hunk 7 above)
Comment (JeremieSamson) [reply assets/sass/components/_homepages.scss]: Tes please :)

File: assets/sass/components/_dashboard.scss (hunk 4 above)
Comment (JeremieSamson) [reply assets/sass/components/_dashboard.scss]: Ok

Comment (JeremieSamson) [reply assets/sass/components/_dashboard.scss]: Ok

File: assets/sass/_vendors.scss (hunk 2 above)
Comment (matthieu-fesselier) [reply assets/sass/_vendors.scss]: For example if a developer wants to use the bootstrap badges, popovers, toasts, ... he has to import the module before. Now it's not imported to make the CSS file a bit lighter. 
The commented part shows the available modules so the developer just has to take a look here to see what's available and import it. All the modules available are not described in the bootstrap doc, so in my opinion, it's easier to show it here

Comment (JeremieSamson) [reply assets/sass/_vendors.scss]: alright, let's keep them then ;)
